### PR TITLE
[CONTINT-4081] fix loading language ttl

### PR DESCRIPTION
**Pull request by adel121**
### What does this PR do?

This PR fixes loading language ttl setting in the language detection handler in the cluster agent. 

### Motivation

In the cluster agent, we use lazy loading to load owners languages in the language detection handler. This is done [here](https://github.com/DataDog/datadog-agent/blob/e68edd16b9de6377854ccb6aa5762db80241890b/cmd/cluster-agent/api/v1/languagedetection/language_detection.go#L51-L69).

However, in this lazy loading, we are also initialising the language ttl for the cleanup process ([here](https://github.com/DataDog/datadog-agent/blob/e68edd16b9de6377854ccb6aa5762db80241890b/cmd/cluster-agent/api/v1/languagedetection/language_detection.go#L54)).

Consequently, as language ttl is accessed for the first time before owners languages, it is given the 0 value of time, which results in a wrong language ttl for the languages detected for the very first container. 

language ttl should have a separate lazy loading.

### Describe how to test/QA your changes

Deploy the agent with APM SSI and language detection enabled, and set the cleanup period retry to low duration in order not to wait for too long:

```
datadog:
  apiKeyExistingSecret: datadog-secret
  appKeyExistingSecret: datadog-secret
  kubelet:
    tlsVerify: false

  # datadog.apm.instrumentation.enabled
  apm:
    instrumentation:
      enabled: true

agents:
  telemetry: enabled
  containers:
    agent:
      env:
        - name: DD_LANGUAGE_DETECTION_REPORTING_REFRESH_PERIOD
          value: "25s"

clusterAgent:
  enabled: true
  admissionController:
    enabled: true
    mutateUnlabelled: true
  env:
    - name: DD_CLUSTER_AGENT_LANGUAGE_DETECTION_CLEANUP_PERIOD
      value: "3m"
```

Create a separate namespace for your application

```
kubectl create namespace application
```

Create a deployment with two containers, one having a regular name, and one having a super long name.
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: python-and-ruby-deployment
  namespace: application
  labels:
    app: user-app
spec:
  replicas: 2
  selector:
    matchLabels:
      app: user-app
  template:
    metadata:
      labels:
        app: user-app
    spec:
      containers:
        - name: some-container
          image: python:3.7 # Replace with your Python image
          command: ["python3", "-u", "your-python-script.py"]
          volumeMounts:
            - name: python-script-volume
              mountPath: /your-python-script.py
              subPath: your-python-script.py
      volumes:
        - name: python-script-volume
          configMap:
            name: python-script-configmap
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: python-script-configmap
  namespace: application
data:
  your-python-script.py: |
    while True:
      pass
```

Wait for about 6 minutes, and check that the language was not removed from workloadmeta. 
Without this fix, the language would be removed because the language ttl is not initialised properly on the first access, which causes the detected languages of the first container to have a negative ttl, which consequently results in removing them on the first cleanup.

### CODE REVIEW

File: cmd/cluster-agent/api/v1/languagedetection/language_detection.go [hunk 1]
```diff
@@ -45,13 +45,14 @@ func InstallLanguageDetectionEndpoints(r *mux.Router, wmeta workloadmeta.Compone
 }
 
 var ownersLanguages OwnersLanguages
-var languageTTL time.Duration
 var ownersLanguagesOnce sync.Once
 
+var languageTTL time.Duration
```
Comment (AliDatadog) [inline cmd/cluster-agent/api/v1/languagedetection/language_detection.go:50]: IMO it's easier to have a struct that we initialize with the right values than to rely on global states.

File: releasenotes-dca/notes/fix-language-ttl-init-a28f3737b37b1b99.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Each section from every release note are combined when the
+# CHANGELOG-DCA.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - Fixed initialisation of language expiration time for detected languages.
```
Comment (hestonhoffman) [inline releasenotes-dca/notes/fix-language-ttl-init-a28f3737b37b1b99.yaml:10]:
```suggestion
  - Fixed initialization of language expiration time for detected languages.
```

**Review by hestonhoffman — APPROVED** [review]
Tiny edit

File: cmd/cluster-agent/api/v1/languagedetection/language_detection.go (hunk 1 above)
Comment (adel121) [reply cmd/cluster-agent/api/v1/languagedetection/language_detection.go]: I fully agree, thanks for the suggestion.

I updated the code.

**Review by jose-manuel-almaza — APPROVED** [review]

File: cmd/cluster-agent/api/v1/languagedetection/handler.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		},
+		wlm:             wlm,
+		ownersLanguages: *newOwnersLanguages(),
+	}
+}
+
+func (handler *languageDetectionHandler) startCleanupInBackground() {
+	// Launch periodic cleanup mechanism
+	go func() {
+		cleanupTicker := time.NewTicker(handler.cfg.cleanupPeriod)
```
Comment (AliDatadog) [inline cmd/cluster-agent/api/v1/languagedetection/handler.go:57]:
```suggestion
		cleanupTicker := time.NewTicker(handler.cfg.cleanupPeriod)
		defer cleanupTicker.Stop()
```

File: cmd/cluster-agent/api/v1/languagedetection/handler.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+)
+
+type handlerConfig struct {
+	enabled       bool
+	languageTTL   time.Duration
+	cleanupPeriod time.Duration
+}
+
+type languageDetectionHandler struct {
+	ownersLanguages OwnersLanguages
```
Comment (AliDatadog) [inline cmd/cluster-agent/api/v1/languagedetection/handler.go:37]:
```suggestion
	ownersLanguages *OwnersLanguages
```
I think a pointer could be useful here.

File: cmd/cluster-agent/api/v1/languagedetection/handler.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+		wlm:             wlm,
+		ownersLanguages: *newOwnersLanguages(),
+	}
+}
+
+func (handler *languageDetectionHandler) startCleanupInBackground() {
+	// Launch periodic cleanup mechanism
+	go func() {
+		cleanupTicker := time.NewTicker(handler.cfg.cleanupPeriod)
+		for range cleanupTicker.C {
```
Comment (AliDatadog) [inline cmd/cluster-agent/api/v1/languagedetection/handler.go:58]: I think we should replace this by 
```
select {
  case <- cleanupTicker.C:
    ...
  case <- ctx.Done():
    ...
}
```

Comment (adel121) [reply cmd/cluster-agent/api/v1/languagedetection/handler.go:58]: Good catch 👍

**Review by AliDatadog — APPROVED** [review]

**Pull request merged**